app/email-preferences/page.tsx

```
@@ -0,0 +1,555 @@
+        newFrequency: 'none'
+      });
+      return;
+    }
+
+    // Otherwise, apply the selected frequency to all non-required emails in this service
+    const emailsInDomain = getEmailsByDomain(domain);
+    emailsInDomain.forEach(emailType => {
+      if (!emailType.required && emailType.frequencies.includes(frequency)) {
+        updatePreference(emailType.id, frequency);
+      }
+    });
```

Biome (1.9.4)

[error] Prefer for...of instead of forEach.

forEach may lead to performance issues when working with large arrays. When combined with functions like filter or map, this causes multiple iterations over the same type.

(lint/complexity/noForEach)